cpp/models/glct_secir/model.h

```diff
@@ -0,0 +1,364 @@
+
+        // --- Check that the dimensions are consistent. ---
+        if ((Eigen::Index)LctState::template get_num_subcompartments<InfectionState::Exposed>() !=
+            params.template get<StartingProbabilitiesExposed>().rows()) {
+            log_error("Constraint check: Dimension of the parameters does not match the number of subcompartments for "
+                      "the Exposed "
+                      "compartment.");
+            return true;
+        }
+        if ((Eigen::Index)LctState::template get_num_subcompartments<InfectionState::InfectedNoSymptoms>() !=
+            params.template get<StartingProbabilitiesInfectedNoSymptoms>().rows()) {
+            log_error(
+                "Constraint check: Dimension of the parameters does not match the number of subcompartments for the "
+                "InfectedNoSymptoms compartment.");
+            return true;
+        }
+        if ((Eigen::Index)LctState::template get_num_subcompartments<InfectionState::InfectedSymptoms>() !=
+            params.template get<StartingProbabilitiesInfectedSymptoms>().rows()) {
+            log_error(
+                "Constraint check: Dimension of the parameters does not match the number of subcompartments for the "
+                "InfectedSymptoms compartment.");
+            return true;
+        }
+        if ((Eigen::Index)LctState::template get_num_subcompartments<InfectionState::InfectedSevere>() !=
+            params.template get<StartingProbabilitiesInfectedSevere>().rows()) {
+            log_error("Constraint check: Dimension of the parameters does not match the number of subcompartments for "
+                      "the InfectedSevere "
+                      "compartment.");
+            return true;
+        }
+        if ((Eigen::Index)LctState::template get_num_subcompartments<InfectionState::InfectedCritical>() !=
+            params.template get<StartingProbabilitiesInfectedCritical>().rows()) {
+            log_error(
+                "Constraint check: Dimension of the parameters does not match the number of subcompartments for the "
+                "InfectedCritical compartment.");
+            return true;
+        }
+
+        return (params.check_constraints() || this->populations.check_constraints());
+    }
+
+    /**
+     * @brief Evaluates the right-hand-side f of the GLCT dydt = f(y, t).
+     *
+     * The GLCT-SECIR model is defined through ordinary differential equations of the form dydt = f(y, t). 
+     * y is a vector containing number of individuals for each (sub-) compartment.
+     * This function evaluates the right-hand-side f of the ODE and can be used in an ODE solver.
+     *
+     * @param[in] pop The current state of the population in the geographic unit we are considering.
+     * @param[in] y The current state of the model (or a subpopulation) as a flat array.
+     * @param[in] t The current time.
+     * @param[out] dydt A reference to the calculated output.
```

Comment:
```suggestion
     * @param[in] dydt The vector containing the derivatives of all subcompartments. 
```

Reply:
Actually, the param[out] is right, as we just store the result in dydt and do not use any values from this (the required state of the model is stored in pop/y). Moreover, the description is copied from the base class, so that i would like to let it as it is.